### DISCUSSION TITLE
Any reason Laravel's event:cache isn't part of the standard recipe?

### DISCUSSION THREAD

**Discussion by chrisan** [deployphp/deployer#3034] [Help needed]
Hello, just running Enlightn and it warned me about not having an event cache.  The task is already defined in the laravel recipe but now I'm wondering if there's a reason not to do this on every deploy?

https://github.com/deployphp/deployer/blob/master/recipe/laravel.php

```
/**
 * Main deploy task.
 */
desc('Deploys your project');
task('deploy', [
    'deploy:prepare',
    'deploy:vendors',
    'artisan:storage:link',
    'artisan:config:cache',
    'artisan:route:cache',
    'artisan:view:cache',
    'artisan:migrate',
    'deploy:publish',
]);
```

https://laravel.com/docs/9.x/events#event-discovery-in-production

> In production, it is not efficient for the framework to scan all of your listeners on every request. Therefore, during your deployment process, you should run the `event:cache` Artisan command to cache a manifest of all of your application's events and listeners.

**Comment by antonmedv** [comment]
Try to look at file history. Probably it was removed? Do we have other discussions about the same issues?

**Reply by chrisan** [reply]
Only thing I could find was the initial addition of the tasks with the note "The new tasks are not added to the deploy task because automatic event discovery is opt-in functionality in Laravel".   https://github.com/deployphp/deployer/commit/a6ff2eb4c6eb87052f94fdf9d4f9c4aad42d7c9c

This was a new feature in Laravel 5.8.9 patch https://laravel.com/docs/5.8/events#event-discovery but even there they recommend to run the cache for production.  

Running this on an application without any custom events/listeners doesn't seem to cause harm. A file `bootstrap/cache/events.php` is created with the following contents

```
<?php return array (
  'App\\Providers\\EventServiceProvider' => 
  array (
    'Illuminate\\Auth\\Events\\Registered' => 
    array (
      0 => 'Illuminate\\Auth\\Listeners\\SendEmailVerificationNotification',
    ),
  ),
);
```

**Reply by antonmedv** [reply]
Let’s create a pr with adding it?

**Reply by chrisan** [reply]
Sure thing: https://github.com/deployphp/deployer/pull/3035
